xtable-api/src/main/java/org/apache/xtable/model/TableChange.java

```diff
@@ -36,4 +36,7 @@ public class TableChange {
 
   /** The {@link InternalTable} at the commit time to which this table change belongs. */
   InternalTable tableAsOfChange;
+
+  // Commit identifier in source table
+  @Builder.Default String sourceIdentifier = "";
```

Comment:
I think this may be safer without a default and require non-null, what do you think?

Similar note for `InternalSnapshot`